src/main/java/teammates/ui/pagedata/AdminActivityLogPageData.java

```diff
@@ -171,76 +183,54 @@ private boolean shouldExcludeLogEntry(ActivityLogEntry logEntry) {
      * Performs the actual filtering, based on QueryParameters
      * returns false if the logEntry fails the filtering process
      */
-    public ActivityLogEntry filterLogs(ActivityLogEntry logEntry) {
-        if (!logEntry.toShow()) {
-            return logEntry;
-        }
-        
+    public boolean filterLogs(ActivityLogEntry logEntry) {
         if (q == null) {
             if (this.queryMessage == null) {
                 this.queryMessage = "Error parsing the query. QueryParameters not created.";
             }
-            logEntry.setToShow(true);
-            return logEntry;
+            return true;
```

Comment:
It seems contradictory with the method signature "returns false if the logEntry fails the filtering process". Even though it has been done this way before, is this the correct behaviour? If it is, add an comment before this line to explain why.